code/modules/mob/living/simple_animal/friendly/cockroach.dm

```diff
@@ -0,0 +1,58 @@
+
+
+/mob/living/simple_animal/cockroach
+	name = "cockroach"
+	desc = "This station is just crawling with bugs."
+	icon_state = "cockroach"
+	icon_dead = "cockroach"
+	health = 1
+	maxHealth = 1
+	turns_per_move = 5
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 270
+	maxbodytemp = INFINITY
+	pass_flags = PASSTABLE | PASSGRILLE | PASSMOB
+	mob_size = MOB_SIZE_TINY
+	response_help  = "pokes"
+	response_disarm = "shoos"
+	response_harm   = "splats"
+	density = 0
+	ventcrawler = 2
+	gold_core_spawnable = 2
+	var/squish_chance = 50
+
+/mob/living/simple_animal/cockroach/death(gibbed)
+	if(ticker.cinematic) //If the nuke is going off, then cockroaches are invincible. Keeps the nuke from killing them, cause cockroaches are immune to nukes.
+		return
+	if(!ckey || gibbed)// stupid staff of change fucking everything up.
```

Comment:
Its a shitty check put in to make sure it doesn't splat if a player is inside so that player cockroaches can be revived during a wizard round by a wand or staff of ressurection. However, if they were going to get gibbed anyways, splat them.